libs/shared/flagd-core/src/lib/targeting/fractional.ts

```diff
@@ -66,36 +66,45 @@ export function fractionalFactory(logger: Logger) {
   };
 }
 
-function toBucketingList(from: unknown[]): { variant: string; fraction: number }[] {
+function relativeWeight(totalWeight: number, weight: number): number {
+  if (weight == 0) {
+    return 0;
+  }
+  return (weight * 100) / totalWeight;
+}
+function toBucketingList(from: unknown[]): {
+  fractions: { variant: string; fraction: number }[];
+  totalWeight: number;
+} {
   // extract bucketing options
   const bucketingArray: { variant: string; fraction: number }[] = [];
 
-  let bucketSum = 0;
+  let totalWeight = 0;
   for (let i = 0; i < from.length; i++) {
     const entry = from[i];
     if (!Array.isArray(entry)) {
       throw new Error('Invalid bucket entries');
     }
 
-    if (entry.length != 2) {
-      throw new Error('Invalid bucketing entry. Require two values - variant and percentage');
+    if (entry.length == 0) {
```

Comment:
```suggestion
    if (entry.length == 0) {
```
I think this check could be improved a bit - it should be 1-2, 0 or more than 2 are all invalid. This is also validated by schema, but people can bypass schema validation.

Reply:
thank you, improved and added a test case